h/api/views.py

```diff
@@ -208,17 +208,46 @@ def read_jsonld(annotation, request):
 @api_config(route_name='api.annotation', request_method='PUT', permission='update')
 def update(annotation, request):
     """Update the specified annotation with data from the PUT payload."""
+    if request.feature('postgres'):
+        annotation_dict = _update_postgres(annotation, request)
+        legacy_annotation = storage.fetch_annotation(request,
+                                                     annotation.id,
+                                                     _postgres=False)
+        _update_elastic(legacy_annotation, request, notify=False)
+        return annotation_dict
+    return _update_elastic(annotation, request, notify=True)
+
+
+def _update_postgres(annotation, request):
+    schema = schemas.UpdateAnnotationSchema(request,
+                                            annotation.target_uri,
+                                            annotation.groupid)
+    appstruct = schema.validate(copy.deepcopy(_json_payload(request)))
```

Comment:
What's the logic behind calling `copy.deepcopy` here?

1. `.validate()` should not (and AFAICT does not) modify its arguments
2. `_json_payload` [returns a new object every time it is called](https://github.com/Pylons/pyramid/blob/a8cad050d8d0e5095a3bf487132f8d6e315779ec/pyramid/request.py#L236).